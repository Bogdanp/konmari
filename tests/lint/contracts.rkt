#lang racket/base

(require racket/contract/base
         racket/contract/region)

(define/contract (a x y [s 'a])
  (-> [natural-number/c natural-number/c] [(or/c 'a 'b)] void?)
  (void))

(define/contract (b a [b #f])
  (->* (natural-number/c)
       ((or/c #f number?))
       any)
  (void))
