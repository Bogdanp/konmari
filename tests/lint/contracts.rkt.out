/contracts.rkt:6:21:warning:identifier 'x' is never used
/contracts.rkt:6:23:warning:identifier 'y' is never used
/contracts.rkt:6:26:warning:identifier 's' is never used
/contracts.rkt:7:7:warning:contract domain expressions should use round parens; did you mean to use ->*?
/contracts.rkt:7:43:warning:contract domain expressions should use round parens; did you mean to use ->*?
/contracts.rkt:10:21:warning:identifier 'a' is never used
/contracts.rkt:10:21:warning:identifier 'a' shadows an earlier binding
/contracts.rkt:10:24:warning:identifier 'b' is never used
/contracts.rkt:10:24:warning:identifier 'b' shadows an earlier binding
/contracts.rkt:11:8:warning:mandatory domain parens should be square
/contracts.rkt:12:8:warning:optional domain parens should be square
/contracts.rkt:16:1:warning:prefer #f over false/c
